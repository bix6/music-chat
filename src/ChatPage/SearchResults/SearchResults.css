.search-results-div {
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  background-image: linear-gradient(135deg, #fccf31 10%, #f55555 100%);
  overflow-y: auto;
}

.search-results-ol {
  border: 2px solid #9708cc;
  background-image: linear-gradient(20deg, #9708cc 10%, #43cbff 100%);
  max-width: 500px;
  margin: 1rem auto;
  text-align: center;
}

.search-results-ol > .cancel-button {
  margin: 1rem 0;
}

.search-results-ol button {
  margin-bottom: 1rem;
}
