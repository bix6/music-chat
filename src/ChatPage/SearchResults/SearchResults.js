import React from 'react';

class SearchResults extends React.Component {
    static defaultProps = {
        searchResults: []
    }

    createListItems = () => {
        return this.props.searchResults.map((result, i) => {
            return (
                // Storing i in the id so we can access it onClick to determine
                // the li element that was clicked
                <li key={ i } id={ i } onClick={ e => this.props.embedVideo(e.target.parentElement.id) }>
                    <img 
                        alt="video thumbnail"
                        src={ result.thumbnail } />
                    <br />
                    { result.title }
                    <br />
                    { result.channelTitle }
                    <br />
                    <button type="button">Embed</button>
                </li>
            )
        })
    }

    render() {
        return (
            <ol>
                { this.createListItems() }
            </ol>
        )
    }
}

export default SearchResults;