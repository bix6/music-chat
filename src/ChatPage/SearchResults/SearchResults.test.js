import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { BrowserRouter } from "react-router-dom";
import SearchResults from "./SearchResults";

describe("SearchResults Component", () => {
  it("Smoke Test: Renders Default", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <SearchResults />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("Snapshot Test: Default", () => {
    const wrapper = shallow(<SearchResults />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
