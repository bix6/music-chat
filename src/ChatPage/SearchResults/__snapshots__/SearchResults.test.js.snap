// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults Component Snapshot Test: Default 1`] = `
<ol
  className="search-results-ol"
>
  <br />
  <button
    type="button"
  >
    Cancel
  </button>
  <br />
</ol>
`;
